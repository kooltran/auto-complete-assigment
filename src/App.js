import React from "react";
import Layout from "./layout";
import styled from "styled-components";
import MyAutoComplete from "./components/autoComplete/autoComplete";

import { ThemeProvider } from "./provider/themeProvider";

const StyledAutoCompleteContainer = styled.div`
  display: flex;
  justify-content: center;

  @media (max-width: 768px) {
    top: 10%;
    height: 100%;
    align-items: center;
  }
`;

export default function App() {
  return (
    <div className="App">
      <ThemeProvider>
        <Layout>
          <StyledAutoCompleteContainer>
            <MyAutoComplete />
          </StyledAutoCompleteContainer>
        </Layout>
      </ThemeProvider>
    </div>
  );
}
