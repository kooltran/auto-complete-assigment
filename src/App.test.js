import React from "react";
import renderer from "react-test-renderer";
import App from "./App";
// import { render, fireEvent, waitFor, screen } from "@testing-library/react";

test("should render MyAutoComplete correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
