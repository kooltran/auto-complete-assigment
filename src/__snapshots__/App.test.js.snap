// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MyAutoComplete correctly 1`] = `
<div
  className="App"
>
  <div
    className="sc-AxgMl hhZmlf"
  >
    <div
      className="sc-AxhCb ccMvAG"
    >
      <nav
        className="sc-AxjAm bvGeWt"
        open={false}
      >
        <button
          aria-checked={true}
          className="ant-switch ant-switch-checked"
          defaultChecked={true}
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          role="switch"
          type="button"
        >
          <span
            className="ant-switch-inner"
          >
            <span
              className="anticon"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                fill="currentColor"
                focusable="false"
                height="1em"
                width="1em"
              >
                sun.svg
              </svg>
            </span>
          </span>
        </button>
        <div
          className="sc-AxhUy eBsLJQ"
        >
          <a
            className="sc-AxirZ iDhapk"
          >
            Become a host
          </a>
          <a
            className="sc-AxirZ iDhapk"
          >
            Help
          </a>
          <a
            className="sc-AxirZ iDhapk"
          >
            Signup
          </a>
          <a
            className="sc-AxirZ iDhapk"
          >
            Login
          </a>
        </div>
      </nav>
      <button
        className="sc-AxiKw cQLtXv"
        onClick={[Function]}
        open={false}
      >
        <div />
        <div />
        <div />
      </button>
    </div>
    <div
      className="sc-AxheI jgPdNW"
    >
      <div
        className="ant-select ant-select-auto-complete ant-select-single ant-select-show-search"
        data-testid="auto-complete-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "width": 200,
          }
        }
      >
        <div
          className="ant-select-selector"
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <span
            className="ant-select-selection-search"
          >
            <input
              aria-activedescendant="undefined_list_0"
              aria-autocomplete="list"
              aria-controls="undefined_list"
              aria-haspopup="listbox"
              aria-owns="undefined_list"
              autoComplete="off"
              className="ant-select-selection-search-input"
              data-testid="auto-complete-input"
              onChange={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onPaste={[Function]}
              readOnly={false}
              role="combobox"
              style={
                Object {
                  "opacity": null,
                }
              }
              unselectable={null}
              value=""
            />
          </span>
          <span
            className="ant-select-selection-placeholder"
          >
            Try VietNam
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
